### DISCUSSION TITLE
how to delete file without to delete the folder too

### DISCUSSION THREAD

**Discussion by hai-shift** [spatie/laravel-medialibrary#3239] [Q&A]
i put all my images in the same folder

but when i try to delete image is delete the folder with all the other images

```
		-products
		--image1.jpg
		--image2.jpg
		--image3.jpg
ֿֿ
```

i try to delete in two way:
```

				$img_to_delete = $this->media->where('uuid',$uuid)->first();
				$this->product?->deleteMedia($img_to_delete->id);

```

and

		`		$this->media->where('uuid',$uuid)->first()->delete();`

but all the time the delete method delete the products folder with all other images i have.

thanks

**Comment by francoism90** [comment]
This seems normal behavior, you are deleting the (root) model directory.

You could use media collections and/or conversions, they share the same purpose.

**Comment by alikashfi** [comment]
same issue with the products folder!
Spatie mentioned it in the [last paragraph of the docs](https://spatie.be/docs/laravel-medialibrary/v7/advanced-usage/using-a-custom-directory-structure#using-a-custom-directory-structure). but what if we want all product images in /products folder and not /products/{uniqueId}?
